Cubical/HITs/SetQuotients/Properties.agda

```diff
@@ -119,14 +119,32 @@ rec set f feq (squash/ x y p q i j) = set (g x) (g y) (cong g p) (cong g q) i j
   g = rec set f feq
 
 rec2 : isSet C
-  → (f : A → B → C)
-  → (∀ a b c → R a b → f a c ≡ f b c)
-  → (∀ a b c → S b c → f a b ≡ f a c)
-  → A / R → B / S → C
-rec2 set f feql feqr =
-  rec (isSetΠ (λ _ → set))
-    (λ a → rec set (f a) (feqr a))
-    (λ a b r → funExt (elimProp (λ _ → set _ _) (λ c → feql a b c r)))
+     → (f : A → B → C)
+     → (∀ a b c → R a b → f a c ≡ f b c)
+     → (∀ a b c → S b c → f a b ≡ f a c)
+     → A / R → B / S → C
+rec2 {_} {C} {_} {A} {_} {B} {_} {R} {_} {S} set f feql feqr = fun
+  where
+    fun₀ : A → B / S → C
+    fun₀ a [ b ] = f a b
+    fun₀ a (eq/ b c r i) = feqr a b c r i
+    fun₀ a (squash/ x y p q i j) = isSet→SquareP (λ _ _ → set)
+      (λ _ → fun₀ a x)
+      (λ _ → fun₀ a y)
+      (λ i → fun₀ a (p i))
+      (λ i → fun₀ a (q i)) j i
+
+    toPath : ∀ (a b : A) (x : R a b) (y : B / S) → fun₀ a y ≡ fun₀ b y
+    toPath a b rab = elimProp (λ _ → set _ _) λ c → feql a b c rab
+
+    fun : A / R → B / S → C
+    fun [ a ] y = fun₀ a y
+    fun (eq/ a b r i) y = toPath a b r y i
+    fun (squash/ x y p q i j) z = isSet→SquareP (λ _ _ → set)
+      (λ _ → fun x z)
+      (λ _ → fun y z)
+      (λ i → fun (p i) z)
+      (λ i → fun (q i) z) j i
```

Comment:
Wouldn't something like this be simpler:

```
rec2 : isSet C
  → (f : A → B → C)
  → (∀ a b c → R a b → f a c ≡ f b c)
  → (∀ a b c → S b c → f a b ≡ f a c)
  → A / R → B / S → C
rec2 setC f feql feqr [ a ] [ b ] = f a b
rec2 setC f feql feqr [ a ] (eq/ x y z i) = feqr a x y z i
rec2 setC f feql feqr [ a ] (squash/ x y p q i j) =
  setC _ _ (cong (rec2 setC f feql feqr [ a ]) p) (cong (rec2 setC f feql feqr [ a ]) q) i j
rec2 setC f feql feqr (eq/ x y z i) b = {!!}
rec2 setC f feql feqr (squash/ x y p q i j) b =
  setC _ _ (cong (λ x → rec2 setC f feql feqr x b) p) (cong (λ x → rec2 setC f feql feqr x b) q) i j
```

I don't have time to figure out the remaining hole, but it should follow from elimProp I think

Reply:
I prefer the original proof, because the logic used here can be used to convert any call to rec2 to a pattern-matching approach. It's also, at least in my opinion, a lot easier to follow when it doesn't involve a bunch of recursive calls to itself.

Reply:
Ok, as long as it computes properly on `[ a ] [ b ]` I guess it doesn't really matter how the rest is done